### PR TITLE
openssl-3.1.6+quic

### PR DESCRIPTION
**Pull request by tmshort**
Rebase onto openssl-3.1.6.

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by tmshort** [comment]
We generally don't care about FIPS checksums nor CIFuzz.

File: .github/workflows/fips-checksums.yml [hunk 1]
```diff
@@ -73,3 +73,29 @@ jobs:
         with:
           name: fips_checksum
           path: artifact/
+  verify-checksums:
```
Comment (wbl) [inline .github/workflows/fips-checksums.yml:76]: If this is busted and we don't care, should we remove it?

Comment (wbl) [reply .github/workflows/fips-checksums.yml:76]: Actually this seems to have worked?

**Review by wbl — APPROVED** [review]

File: .github/workflows/fips-checksums.yml (hunk 1 above)
Comment (tmshort) [reply .github/workflows/fips-checksums.yml:76]: Still failed, but it's something we can look at: removing some of these CI actions.

**Pull request merged**